File: app/src/main/java/com/dumuzeyn/mp3player/SoundAnalysisState.java
package com.dumuzeyn.mp3player;

enum SoundAnalysisState {
    NOT_ANALYZED,
    QUEUED,
    ANALYZING,
    ANALYZED,
    FAILED;

    static SoundAnalysisState parse(String value) {
        try {
            return valueOf(value == null ? "" : value);
        } catch (IllegalArgumentException error) {
            return NOT_ANALYZED;
        }
    }
}

File: app/src/main/java/com/dumuzeyn/mp3player/SoundAnalysisState.kt
package com.dumuzeyn.mp3player

enum class SoundAnalysisState {
    NOT_ANALYZED,
    QUEUED,
    ANALYZING,
    ANALYZED,
    FAILED;

    companion object {
        @JvmStatic
        fun parse(value: String?): SoundAnalysisState =
            entries.firstOrNull { it.name == value } ?: NOT_ANALYZED
    }
}
